shaders/circle.vertex.glsl

```diff
@@ -0,0 +1,24 @@
+// set by gl_util
+uniform float u_size;
+
+attribute vec2 a_pos;
+
+uniform mat4 u_matrix;
+uniform mat4 u_exmatrix;
+
+varying vec2 v_extrude;
+
+void main(void) {
+    // unencode the extrusion vector that we snuck into the a_pos vector
+    v_extrude = vec2(
+        mod(a_pos.x, 2.0) * 2.0 - 1.0,
+        mod(a_pos.y, 2.0) * 2.0 - 1.0);
```

Comment:
Most glsl operations can work with vectors, so this can be written as `v_extrude = vec2(mod(a_pos, 2.0) * 2.0 - 1.0);`